### PR TITLE
[SYCL][CUDA] Remove pi Event Callback implementation

### PR DESCRIPTION
**Pull request by Ruyk**
Since introduction of host tasks in #1471,
`piEventCallback` and related functionality is not required by the SYCL-RT.
Removing the implementation of this behaviour from the CUDA backend simplifies the submission of operations to streams and increases performance in multiple benchmarks.

Signed-off-by: Ruyman Reyes <ruyman@codeplay.com>

### CODE REVIEW

**Comment by Ruyk** [comment]
@StuartDAdams  / @steffenlarsen  please take a look , this is basically reverting your original work on callbacks!

**Comment by StuartDAdams** [comment]
It is glorious. Good riddance.

File: sycl/plugins/cuda/pi_cuda.cpp [hunk 1]
```diff
@@ -276,13 +276,15 @@ _pi_event::_pi_event(pi_command_type type, pi_context context, pi_queue queue)
...
-  if (is_native_event()) {
+  if (type != PI_COMMAND_TYPE_USER) {
     PI_CHECK_ERROR(cuEventCreate(&evEnd_, CU_EVENT_DEFAULT));
 
     if (queue_->properties_ & PI_QUEUE_PROFILING_ENABLE) {
       PI_CHECK_ERROR(cuEventCreate(&evQueued_, CU_EVENT_DEFAULT));
       PI_CHECK_ERROR(cuEventCreate(&evStart_, CU_EVENT_DEFAULT));
     }
+  } else {
+    cl::sycl::detail::pi::die("User-defined events not implemented");
```
Comment (StuartDAdams) [inline sycl/plugins/cuda/pi_cuda.cpp:287]: Would be good to give more granular message so that the user knows this error is coming from the CUDA backend.

**Comment by steffenlarsen** [comment]
LGTM!

File: sycl/plugins/cuda/pi_cuda.cpp (hunk 1 above)
Comment (Ruyk) [reply sycl/plugins/cuda/pi_cuda.cpp]: Fixed in  adcf8978b525aef45a5ffc888a42722017987c45

**Comment by s-kanaev** [comment]
@Ruyk some cuda tests are failing, please fix.

**Comment by steffenlarsen** [comment]
The test failures don't seem to be caused by this PR. I have made a separate PR to fix the tests: https://github.com/intel/llvm/pull/1763

**Comment by bader** [comment]
> The test failures don't seem to be caused by this PR. I have made a separate PR to fix the tests: #1763

Any ideas how this test failure might be introduced? I thought that buildbot check should had caught such regressions.

**Pull request merged**